x-pack/plugins/uptime/public/components/overview/kuery_bar/typeahead/search_type/search_type.tsx

```diff
@@ -0,0 +1,139 @@
+  kqlSyntax: boolean;
+  setKqlSyntax: (val: boolean) => void;
+}
+
+export const SearchType = ({ kqlSyntax, setKqlSyntax }: Props) => {
+  const {
+    services: { docLinks },
+  } = useKibana();
+
+  const [getUrlParams, updateUrlParams] = useUrlParams();
+
+  const { query, search } = getUrlParams();
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const onButtonClick = () => setIsPopoverOpen((prevState) => !prevState);
+
+  const closePopover = () => setIsPopoverOpen(false);
+
+  useEffect(() => {
+    if (kqlSyntax && query) {
+      updateUrlParams({ query: '' });
+    }
+
+    if (!kqlSyntax && search) {
+      updateUrlParams({ search: '' });
+    }
+    // don't include kqlSyntax
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isPopoverOpen, query, search, updateUrlParams]);
+
+  const button = kqlSyntax ? (
+    <EuiButtonEmpty
+      data-test-subj="syntaxChangeToSimple"
+      onClick={onButtonClick}
+      aria-label={CHANGE_SEARCH_BAR_SYNTAX_SIMPLE}
+      title={CHANGE_SEARCH_BAR_SYNTAX_SIMPLE}
+    >
+      KQL
+    </EuiButtonEmpty>
+  ) : (
+    <BoxesVerticalIcon
+      type="boxesVertical"
+      onClick={onButtonClick}
+      data-test-subj="syntaxChangeToKql"
+      aria-label={CHANGE_SEARCH_BAR_SYNTAX}
+      title={CHANGE_SEARCH_BAR_SYNTAX}
+    />
+  );
+
+  return (
+    <>
+      <EuiPopover
+        button={button}
+        isOpen={isPopoverOpen}
+        closePopover={closePopover}
+        ownFocus={true}
+        anchorPosition="downRight"
+      >
+        <div style={{ width: '360px' }}>
+          <EuiPopoverTitle>{SYNTAX_OPTIONS_LABEL}</EuiPopoverTitle>
+          <EuiText>
+            <p>
+              <FormattedMessage
+                id="xpack.uptime.queryBar.syntaxOptionsDescription"
+                defaultMessage="The {docsLink} (KQL) offers a simplified query
+              syntax and support for scripted fields. KQL also provides autocomplete if you have
+              a Basic license or above. If you turn off KQL, Uptime
+            uses simple wildcard search against {searchField} fields."
+                values={{
+                  docsLink: (
+                    <EuiLink href={docLinks!.links.query.kueryQuerySyntax} target="_blank" external>
+                      <FormattedMessage
+                        id="xpack.uptime.query.queryBar.kqlFullLanguageName"
+                        defaultMessage="Kibana Query Language"
+                      />
+                    </EuiLink>
+                  ),
+                  searchField: <strong>Monitor Name, ID, Url</strong>,
+                }}
+              />
+            </p>
+          </EuiText>
+          <EuiSpacer />
+          <EuiFormRow label="Kibana Query Language" id="asdf" hasChildLabel={false}>
+            <EuiSwitch
+              name="switch"
+              label={kqlSyntax ? 'On' : 'Off'}
+              checked={kqlSyntax}
+              onChange={() => setKqlSyntax(!kqlSyntax)}
```

Comment:
This should be wrapped in `useCallback`, as a new function is being created on each rerender here.

Reply:
Also, when you create useCallback, you can use functional set state here to alieviate the need for including `kqlSyntax` as a dependency

`useCallback(() => setKqlSyntax((kqlSyntax) => !kqlSyntax), [setKqlSyntax])`